nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/AbstractGCPProcessor.java

```diff
@@ -43,7 +43,8 @@
             .Builder().name("gcp-project-id")
             .displayName("Project ID")
             .description("Google Cloud Project ID")
-            .required(true)
```

Comment:
in what case the project id could be null?

Reply:
For example to download public buckets like this one:
https://console.cloud.google.com/storage/browser/gcp-public-data-landsat/?_ga=2.37550372.-565124473.1518597165

I created an issue for that a while ago: 
https://issues.apache.org/jira/browse/NIFI-4933

Reply:
OK get it. So it's for the other JIRA. Good to mention it to be sure we update the JIRA when we merge this one.